Mobile/ShopLens/Droid/Activities/CameraActivity.cs

```diff
@@ -230,6 +238,17 @@ public void OnResults(Bundle results)
             }
         }
 
+        public override void OnRequestPermissionsResult(int requestCode, string[] permissions, Permission[] grantResults)
+        {
+            if (requestCode == REQUEST_PERMISSION)
```

Comment:
Is `REQUEST_PERMISSION` referring to any permission request or external storage write permission request? Because on line 247 we request only WriteExternalStorage permissions

Reply:
Any, it's part of the global request Id enum.

Reply:
So `OnRequestPermissionsResult` gets called when we have a permission result.

Then we check if the request was to get permissions (always true?)

Then we say, "if you requested permissions and it failed, request this specific WriteExternalStorage permission".

Does this seem ok?

Reply:
We check whether the permission request Id was of a request that had a specific Id attached to it, because if we had a couple of permission requests in the same activity they could have different Ids. So it's not always true, if we use different Ids.

Reply:
The Camera and Microphone permissions are requested in MainActivity, I removed the Write permission request from there so that the handling of the MkDirs thrown exception wasn't obsolete or useless. If you actually add the storage permission to MainActivity, there's no way for the user to get past to CameraActivity and have no storage permission allowed for our app right now.

Reply:
So since this exception handling is set up so only for the second assignment, we'll just get rid of it later. It does seem ok, unless you have any other ideas where we could handle exceptions in a meaningful way.